modular_skyrat/modules/conveyor_sorter/code/conveyor_sorter.dm

```diff
@@ -145,3 +143,59 @@
 		"conveysorter",
 	)
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
+
+/obj/item/conveyor_sorter/improved
+	name = "improved conveyor sorter lister"
+	desc = "A tool that is used to not only create the conveyor sorters, but give lists to the conveyor sorters. This one has an increased capacity of eight."
+	icon_state = "lister_improved"
+	max_sorters = 8
+
+/obj/item/conveyor_sorter/improved/examine(mob/user)
+	. = ..()
+	. += span_notice("This version of the sorter allows up to eight sorters to be placed.")
+
+/obj/item/conveyor_sorter/improved/attack_self(mob/user, modifiers)
+	if(length(spawned_sorters) >= max_sorters)
+		to_chat(user, span_warning("You may only have [max_sorters] spawned conveyor sorters!"))
+		return
+	var/obj/effect/decal/cleanable/conveyor_sorter/improved/new_cs = new /obj/effect/decal/cleanable/conveyor_sorter/improved(get_turf(src))
```

Comment:
Make the type of `conveyor_sorter` here a variable on the `/obj/item/conveyor_sorter` class, and then change it in the improved version, so instead of overriding this whole proc, you just change one variable and that's it.